### PR TITLE
[31974] Simplify instance color theme changes

### PR DESCRIPTION
**Pull request by HDinger**
This PR aims to simplify the change of colour schemes of an OpenProject instances with EE token. Therefore the view itself is made a bit more intuitive. Further predefined colour themes shall be provided.

### ToDo
- [x] Improve UX of design variable list 
  - [x] Change order of changeable design variables so that logical blocks are shown together
  - [x] Provide a natural language name for each variable
  - [x] Use all already available help texts for design variables
- [x] Provide predefined color themes
  - [x] Show a select field with themes and hide the design variable list under "Advanced settings"
  - [x] Changing the color theme changes the variables when clicking on Save (Out of scope: Instead change when choosing a theme)
  - [x] Recognise the used theme, to show the correct one in the list.
  - [x] When changing a colour manually, the select field with themes should show a placeholder

### CODE REVIEW

**Comment by HDinger** [comment]
Ticket: 
https://community.openproject.com/projects/openproject/work_packages/31974/activity

File: config/locales/en.yml [hunk 1]
```diff
@@ -37,21 +37,39 @@ en:
...
-        content_link_color: "Font color of most of the links."
-        primary_color: "Main color."
-        primary_color_dark: "Typically a darker version of the main color used for hover effects."
-        header_bg_color: "Background color of the header."
-        header_item_bg_hover_color: "Background color of clickable header items when hovered with the mouse."
-        header_item_font_color: "Font color of clickable header items."
-        header_item_font_hover_color: "Font color of clickable header items when hovered with the mouse."
-        header_border_bottom_color: "Thin line under the header. Leave this field empty if you don't want any line."
-        main_menu_bg_color: "Left side menu's background color."
+        alternative-color: "Strong accent color, typically used for most the important button on a screen."
```
Comment (ulferts) [inline config/locales/en.yml]: typo: `most the important` -> `the most important`

File: app/views/custom_styles/show.html.erb [hunk 2]
```diff
@@ -139,12 +156,12 @@ See docs/COPYRIGHT.rdoc for more details.
...
-    <fieldset class="form--fieldset">
-      <legend class="form--fieldset-legend"><%= I18n.t('admin.custom_styles.custom_colors') %></legend>
+    <fieldset class="form--fieldset -collapsible collapsed">
+      <legend class="form--fieldset-legend"><%= I18n.t(:label_advanced_settings) %></legend>
 
       <% DesignColor.setables.each do |design_color|  %>
         <div class="form--field -required">
-          <label class="form--label"><%= design_color.variable %>:</label>
+          <label class="form--label"><%= I18n.t("admin.custom_styles.colors.#{design_color.variable}") %>:</label>
           <span class="form--field-container">
```
Comment (ulferts) [inline app/views/custom_styles/show.html.erb:165]: Can we please use the chance to move the color representation right next to the input field? 

E.g. by adding
```
            <div class="form--field-affix">
              <%= icon_for_color(OpenStruct.new(variable: design_color.variable,
                                                hexcode: design_color.get_hexcode),
                                 data: { target: "#design_colors_" + design_color.variable }) %>
            </div>
```
as the first child of the `form--field-container` we get:

image

**Review by ulferts — CHANGES_REQUESTED** [review]
A solid improvement @HDinger :+1: 

The only beef I have with the change is that when a custom theme is selected, the user will always have to first open the collapsed colors section to see/edit. How about we change this to be open by default if a custom theme is selected?

**Review by ulferts — APPROVED** [review]
Looks good now @HDinger, good to merge once travis and codeclimate agree.

**Pull request merged**